Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -702,18 +702,38 @@ public class ProgramBuilder {
         //   Otherwise copy the whole block including its content
         var requiredInputs = VariableSet()
 
+        // A Set of variables that have yet to be included in the slice
+        var remainingInputs = VariableSet()
+
+        // A stack of contexts that are required by the instruction in the slice
+        var requiredContextStack = [Context.empty]
+
         // Helper function to add an instruction, or possibly multiple instruction in the case of blocks, to the slice.
         func add(_ instr: Instruction, includeBlockContent: Bool = false) {
             guard !slice.contains(instr.index) else { return }
 
             func internalAdd(_ instr: Instruction) {
+                if !remainingInputs.isEmpty {
```

Comment:
is the check here and below required? Isn't just `remainingInputs.subtract(instr.allOutputs)` enough?